redaxo/src/addons/media_manager/lib/managed_media.php

```diff
@@ -341,10 +341,8 @@ public function setSourcePath($path)
         $this->sourcePath = $path;
 
         $this->asImage = false;
-        if (!isset($this->image['src'])) {
-            return;
-        }
-        if (!is_resource($this->image['src'])) {
+        // prepared for propper use PHP7 "is_resource($this->image['src'])" and PHP8 "$this->image['src'] instanceof \GdImage)"
+        if (!is_resource($this->image['src']) && !($this->image['src'] instanceof \GdImage)) { 
             return;
         }
         imagedestroy($this->image['src']);
```

Comment:
Ich hatte das Addon in Vorbereitung für PHP 8 schon mal nach `is_resource` abgesucht und bin da schon über diese Stelle gestolpert.
Mit dem if oben wird aber ja nur der `imagedestroy`-Aufruf hier umgangen. Und `imagedestory` hat seit PHP 8 keine Auswirkungen mehr, siehe Doku: https://www.php.net/manual/de/function.imagedestroy.php
Deswegen macht es nichts, wenn in php 8 der `is_resource`-Aufruf oben `false` liefert und dann `imagedestory` nicht aufgerufen wird. Daher habe ich die Stelle so belassen.

Ich würde den Code daher so belassen, wie er ist. Aber wir können gerne einen erklärenden Kommentar aufnehmen, damit man nicht erneut drüber stolpert.

Reply:
Wenn in nicht mehr allzu ferner Zukunft REDAXO auf PHP 8 als Mindestversion aufsetzt, wäre ja zuvor eine Code-Bereinigugn sinnvoll. Damit man die Stellen schneller findet, ist ein "Finde"-Kommentar imho auf jeden Fall sinnvoll. Habt ihr schon ein Codewort für die Suche?

Man könnte ja bei der Gelegenheit auch die Frage umdrehen ...  
```php
// <<<PHP7 only
if (is_resource($this->image['src'])) {
        imagedestroy($this->image['src']);
}
// >>>
```

Reply:
Hab es entsprechend geändert. Danke für die Erklärungen und Hinweise